app/merginprojectstatusmodel.cpp

```diff
@@ -0,0 +1,127 @@
+#include "merginprojectstatusmodel.h"
+#include "geodiffutils.h"
+#include "inpututils.h"
+
+MerginProjectStatusModel::MerginProjectStatusModel( LocalProjectsManager &localProjects, QObject *parent )
+  : QAbstractListModel( parent )
+  , mLocalProjects( localProjects )
+{
+}
+
+int MerginProjectStatusModel::rowCount( const QModelIndex &parent ) const
+{
+  return mItems.length();
+}
+
+QHash<int, QByteArray> MerginProjectStatusModel::roleNames() const
+{
+  QHash<int, QByteArray> roleNames = QAbstractListModel::roleNames();
+  roleNames[Status] = "fileStatus";
+  roleNames[Text] = "itemText";
+  roleNames[Filename] = "filename";
+  roleNames[Inserts] = "inserts";
+  roleNames[Deletes] = "deletes";
+  roleNames[Updates] = "updates";
+  roleNames[Section] = "section";
+  return roleNames;
+}
+
+QVariant MerginProjectStatusModel::data( const QModelIndex &index, int role ) const
+{
+  int row = index.row();
+  if ( row < 0 || row >= mItems.count() )
+    return QVariant();
+
+  ProjectStatusItem item = mItems.at( row );
+
+  switch ( role )
+  {
+    case Status: return item.status;
+    case Text: return item.text;
+    case Filename: return item.filename;
+    case Inserts: return item.inserts;
+    case Deletes: return item.deletes;
+    case Updates: return item.updates;
+    case Section: return item.section;
+  }
+  return QVariant();
+
+}
+
+void MerginProjectStatusModel::insertIntoItems( const QSet<QString> &files, const ProjectChangelogStatus &status, const QString &projectDir )
+{
+  for ( QString file : files )
+  {
+    if ( !MerginApi::isInIgnore( QFileInfo( projectDir + "/" + file ) ) )
+    {
+      ProjectStatusItem item;
+      item.status = status;
+      item.text = file;
+      item.section = "Changes";
+      mItems.append( item );
+    }
+  }
+}
+
+void MerginProjectStatusModel::infoProjectUpdated( const ProjectDiff &projectDiff, const QString &projectDir )
+{
+  beginResetModel();
+  mItems.clear();
+
+  insertIntoItems( projectDiff.localUpdated, ProjectChangelogStatus::Updated, projectDir );
+  insertIntoItems( projectDiff.localAdded, ProjectChangelogStatus::Added, projectDir );
+  insertIntoItems( projectDiff.localDeleted, ProjectChangelogStatus::Deleted, projectDir );
+
+  for ( QString file : projectDiff.localUpdated )
+  {
+    if ( MerginApi::isFileDiffable( file ) )
+    {
+      QString summaryJson = GeodiffUtils::diffableFilePendingChanges( projectDir, file, true );
+      if ( summaryJson.startsWith( "ERROR" ) )
+      {
+        InputUtils::log( "MerginProjectStatusModel", QString( "Diff summary JSON for %1 in %2 has an error." ).arg( projectDir ).arg( file ) );
```

Comment:
could we get also an item for the error in GUI so it is visible to the user that diff went wrong?